pre_commit_hooks/pretty_format_json.py

```diff
@@ -111,24 +112,24 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
                 contents, args.indent, ensure_ascii=not args.no_ensure_ascii,
                 sort_keys=not args.no_sort_keys, top_keys=args.top_keys,
             )
-
-            if contents != pretty_contents:
-                if args.autofix:
-                    _autofix(json_file, pretty_contents)
-                else:
-                    print(
-                        get_diff(contents, pretty_contents, json_file),
-                        end='',
-                    )
-
-                status = 1
         except ValueError:
             print(
                 f'Input File {json_file} is not a valid JSON, consider using '
                 f'check-json',
             )
             return 1
 
+        if contents != pretty_contents:
+            if args.autofix:
+                _autofix(json_file, pretty_contents)
+            else:
+                diff_output = get_diff(
+                    contents, pretty_contents, json_file,
+                )
+                sys.stdout.buffer.write(diff_output.encode('utf-8'))
```

Comment:
UTF-8 is the default encoding in all versions supported, this can be `.encode()`